ci-scripts/linux/local-build/linux_build.sh

```diff
@@ -0,0 +1,559 @@
+            2)
+                echo "$(_msg CLONING_REPOSITORY)"
+                git clone https://github.com/tahoma2d/tahoma2d "$TAHOMA_DIR"
+                if [ -d "$TAHOMA_DIR" ]; then
+                    echo "$(_msg CLONING_COMPLETED_CONTINUING)"
+                    sleep 2
+                    return
+                else
+                    echo "$(_msg CLONING_ERROR_RETRY)"
+                fi
+                ;;
+			3)
+				read -r -p "$(_msg INTRO_PATH) " USER_DIR
+				USER_DIR=$(echo "$USER_DIR" | sed 's/^"//;s/"$//') # Remove unnecessary quotes
+				USER_DIR=$(eval echo "$USER_DIR") # Expand tilde (~) and environment variables
+
+				# If the user provided the parent directory, correct it automatically
+				if [ -d "$USER_DIR/tahoma2d" ]; then
+					USER_DIR="$USER_DIR/tahoma2d"
+				fi
+
+				if [ -d "$USER_DIR" ]; then
+					export TAHOMA_DIR="$USER_DIR"
+					export STUFF_DIR="$TAHOMA_DIR/stuff"
+					export LIBTIFF_DIR="$TAHOMA_DIR/thirdparty/tiff-4.2.0"
+					export TOONZ_DIR="$TAHOMA_DIR/toonz"
+					export BUILD_DIR="$TOONZ_DIR/build"
+					export BIN_DIR="$BUILD_DIR/bin"
+					export SOURCES_DIR="$TOONZ_DIR/sources"
+
+					echo "$(_msg TAHOMA2D_FOLDER_FOUND_AT) \"$TAHOMA_DIR\""
+					sleep 2
+					return
+				else
+					echo "$(_msg DIRECTORY_NOT_FOUND_RETRY)"
+					sleep 2
+				fi
+				;;
+
+            4)
+                exit 1
+                ;;
+            *)
+                echo "$(_msg INVALID_OPTION_RETRY)"
+                ;;
+        esac
+    done
+}
+
+function _stuff() {
+
+    mkdir -p "$USER_CONFIG_DIR"
+    # Check if the 'stuff' directory exists before copying
+    if [ -d "$STUFF_DIR" ]; then
+        cp -rf "$STUFF_DIR" "$USER_CONFIG_DIR/"
+    fi
+}
+
+
+function _systemVar() {
+
+    mkdir -p "$USER_CONFIG_DIR"
+
+    cat << EOF > "$USER_CONFIG_DIR/SystemVar.ini"
+[General]
+TAHOMA2DROOT="$USER_CONFIG_DIR/stuff"
+TAHOMA2DPROFILES="$USER_CONFIG_DIR/stuff/profiles"
+TAHOMA2DCACHEROOT="$USER_CONFIG_DIR/stuff/cache"
+TAHOMA2DCONFIG="$USER_CONFIG_DIR/stuff/config"
+TAHOMA2DFXPRESETS="$USER_CONFIG_DIR/stuff/fxs"
+TAHOMA2DLIBRARY="$USER_CONFIG_DIR/stuff/library"
+TAHOMA2DPROJECTS="$USER_CONFIG_DIR/stuff/projects"
+TAHOMA2DSTUDIOPALETTE="$USER_CONFIG_DIR/stuff/studiopalette"
+EOF
+}
+
+function _libTiff() {
+    cd "$LIBTIFF_DIR"
+    ./configure --with-pic --disable-jbig --disable-webp
+    make -j$(nproc)
+    cd "$TAHOMA_DIR"  # Return to the base directory.
+}
+
+function _build() {
+    if [ ! -d "$SOURCES_DIR" ]; then
+        clear
+        echo -e "\n\n$(_msg SOURCES)\n\n"
+        exit 1
+    fi
+    
+    mkdir -p "$BUILD_DIR"   
+
+    cd "$BUILD_DIR"  # Make sure the compilation is done in the correct directory.
+
+    if command -v dnf >/dev/null 2>&1 || command -v yum >/dev/null 2>&1; then
+        cmake "$SOURCES_DIR" -DCMAKE_INSTALL_PREFIX="$BUILD_DIR" -DSUPERLU_INCLUDE_DIR=/usr/include/SuperLU
+    else
+        cmake "$SOURCES_DIR" -DCMAKE_INSTALL_PREFIX="$BUILD_DIR"
```

Comment:
This command failed for me, and it took way too long to figure out what went wrong because of all the `clear` and `dialog` that hid the error. You need to check the return code of most commands that are executed to see if they failed and decide whether you need to abandon the script.